### DISCUSSION TITLE
p:dataTable: Custom filter for List-valued fields?

### DISCUSSION THREAD

**Discussion by mdzaebel** [primefaces/community#1703] [PrimeFaces]
I have a column, that is List-valued. I want a filter in the form, that selects intersection of the List and selected colors:
Example for multi-selection with a single-value-field:
```
<p:column filterBy="#{car.color}" headerText="Color" filterMatchMode="in">
            <f:facet name="filter">
                <p:selectCheckboxMenu label="Colors" onchange="PF('carsTable1').filter()">
                    <f:selectItems value="#{view.colors}" />
                </p:selectCheckboxMenu>
            </f:facet>
            <h:outputText value="#{car.color}" />
</p:column>
```
I'm using LazyDataModel

**Comment by Rapster** [comment]
what is the question?

**Reply by mdzaebel** [reply]
The question is, how to **implement** the multi-valued filter for a multi-valued field.

**Reply by Rapster** [reply]
what do you get in your lazy callback, if you do: `filterBy.get("color").getFilterValue()` ?

**Reply by mdzaebel** [reply]
I've not implemented it, but need to use LazyDataModel, as there is a huge amount of data. My sample above is from documentation, but there is no example with a List-valued field and a p:selectCheckboxMenufilter. So even without a LazyDataModel iplementation, an answer would help much.

**Reply by Rapster** [reply]
`FilterMeta#getFilterValue` should return your selected colors

**Comment by mdzaebel** [comment]
I'll try the following:
```
<p:dataTable value="#{view.rows}" var="v">
    <p:column filterBy="#{v.list}" filterMatchMode="custom" filterFunction="#{view.filterByList}"> ... </p:column>
</p:dataTable>
```

**Reply by mdzaebel** [reply]
Seems, my filterFunction is not applied, as no output is printed:

```
public boolean filterIt(Object value, Object filter, Locale locale) {
            System.out.println("filterIt:"+value+","+filter);
            return true;
}
```
```
<p:dataTable value="#{view.rows}" var="v">
    <p:column filterBy="#{v.list}" filterFunction="#{view.filterIt}"> ... </p:column>
</p:dataTable>
```
It's documented exactly this way at https://primefaces.github.io/primefaces/14_0_0/#/components/datatable?id=lazy-loading
